### ISSUE TITLE
unit file specifiers: add specifier for final component of the suffix

### ISSUE THREAD

**Issue by adsk-mcnultt** [systemd/systemd#14244]
**Is your feature request related to a problem? Please describe.**
I'm trying to use systemd to manage multiple singular worker processes. I would like to define a `resque-worker@.service` template file that would accept 
`resque-worker@queuename.service` commands, however this means I can't have two processes on a single host taking from a singular queue, as I use the instance name specifier to do logic for pid and log files. 

**Describe the solution you'd like**
My configuration for these processes is programatic and within my template file I make use of the instance name specifier, the portion of the call AFTER the @, in this case queuename. I would like to be able to call systemd using `resque-worker@queuename-uniqueid.service` and reference uniqueid within the template file. 

A specifier "%j" exists as "Final component of the prefix." It would make sense the suffix offered the same sort of ability. 

**Describe alternatives you've considered**
I started off trying to run some bash that'd split this all up, and additionally had some logic that would map a counted value to a file containing environment files, but given the project's feeling towards environmentfile I feel that's not suitable.

**Comment by anitazha** [comment]
Components in the prefix have significance when doing overrides. But I think instances are supposed to be unique to a prefix so components might not make sense. Maybe someone else can comment more definitively.

Just a suggestion, but a work around could be to have a systemd generator to write out the different templates with a unique prefix name and constant content (e.g. `resque-worker-queuename@uniqueid`) and use the existing facilities to get the queue name and instance.

**Comment by Capt-Tom** [comment]
I agree with your workaround, it would work but slides past the point of a template in my opinion. If I place the queue name in the prefix I end up with dozens of separate service templates. That's a bit messier than being able to pass it as a specifier later. 

For the time being I'm going to break this logic into a deeper portion of the project.

**Comment by ArmandoZepeda** [comment]
I'm trying to do same action with the instance suffix. 

- I want to use instance prefix as root folder name for different components it may have. 
- Then, the scaped instance name suffix will be relative path to instance working directory.
- Inside the working directory will have the exec bin(with name same as last dir name). Some config files and assets required for the instance may be found in this directory too.

This way allows to have easy clasification and documentation of several projects and their own internal components.

**This is an example:**
Templates:
```
my_system_core@.service
my_system_tools@.service
```

Instances:
```
my_system-core@collector-downloader
my_system-core@collector-storage
my_system-core@reports-csv_builder
my_system-tools@stats-mail_sender
my_system-tools@stats-ftp_uploader
```

Exec bin files:
```
/my_system/core/collector/downloader/downloader
/my_system/core/collector/storage/storage
/my_system/core/reports/csv_builder/csv_builder
/my_system/tools/stats/mail_sender/mail_sender
/my_system/tools/stats/ftp_uploader/ftp_uploader
```

Using this structure I need to provide the valid exec bin file to ExecStart.
Something like below is not valid:
```
ExecStart=/%P/%i/%i
ExecStart=/%P/%I
```

I would prefer not to use a wrapper to achieve this.

**Comment by mcassaniti** [comment]
Even something simple like the following using a final suffix would be useful. The example below copies a file for an absolute path to an explicit location using just the final suffix (lets call it `%F` in this example).

Path unit, something like `my-sync@.path`
```INI
[Path]
# %I is an absolute path to the file to watch
PathChanged = "%I"

[Install]
WantedBy = paths.target
```

Service unit, something like `my-sync@.service`
```INI
[Unit]
Description = Copy changed file to destination directory
After       = network.target

[Service]
Type          = oneshot
ExecStart     = /usr/bin/cp "%I" "/path/to/dest-dir/%F"
```

If @poettering is happy with a change like this then I'll see if I can get a rough PR together.
